flocker/control/_protocol.py

```diff
@@ -350,7 +351,12 @@ def cancel(self):
         """
         Cancel the delayed call to this ``Timeout``'s ``action``.
         """
-        self._delay_call.cancel()
+        try:
+            self._delay_call.cancel()
+        except AlreadyCalled:
+            # This Timeout may have triggered protocol.abortConnection which
+            # will attempt to cancel...this timeout...
+            pass
```

Comment:
The error occurs as a result of  `{Control,Agent}AMP.connectionLost` method. 
And unless that succeeds the `ReconnectingClientFactory` doesn't get notified.